api/src/test/java/org/pmiops/workbench/utils/mappers/AuditLogEntryMapperTest.java

```diff
@@ -0,0 +1,162 @@
+package org.pmiops.workbench.utils.mappers;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import static org.pmiops.workbench.actionaudit.auditors.ActionAuditTestConfig.ACTION_ID;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import org.joda.time.DateTime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.model.AuditAction;
+import org.pmiops.workbench.model.AuditAgent;
+import org.pmiops.workbench.model.AuditEventBundle;
+import org.pmiops.workbench.model.AuditEventBundleHeader;
+import org.pmiops.workbench.model.AuditLogEntry;
+import org.pmiops.workbench.model.AuditTargetPropertyChange;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class AuditLogEntryMapperTest {
+
+  public static final long AGENT_ID = 101L;
+  public static final String USER_AGENT_TYPE = "USER";
+  public static final String AGENT_USERNAME = "paul@beatles.info";
+  public static final long TARGET_ID = 202L;
+  public static final String ACTION_TYPE_DELETE = "DELETE";
+  public static final String TARGET_TYPE_WORKSPACE = "WORKSPACE";
+  public static final String TARGET_PROPERTY = "title";
+  public static final String PREVIOUS_VALUE = "District 5";
+  public static final String NEW_VALUE = "The Mighty Ducks";
+  public static final String LOGIN_ACTION_TYPE = "LOGIN";
+  public static final DateTime EVENT_TIME = new DateTime(1579013840545L);
+  public static final String WORKBENCH_TARGET_TYPE = "WORKBENCH";
+  @Autowired private AuditLogEntryMapper auditLogEntryMapper;
+
+  @TestConfiguration
+  @Import({AuditLogEntryMapperImpl.class})
+  public static class Config {}
+
+  @Test
+  public void testLogEntryToAgent() {
+    final AuditLogEntry logEntry =
+        new AuditLogEntry()
+            .actionType("CREATE")
+            .agentId(AGENT_ID)
+            .agentType(USER_AGENT_TYPE)
+            .agentUsername(AGENT_USERNAME);
+    final AuditAgent auditAgent = auditLogEntryMapper.logEntryToAgent(logEntry);
+    assertThat(auditAgent.getAgentId()).isEqualTo(AGENT_ID);
+    assertThat(auditAgent.getAgentType()).isEqualTo(USER_AGENT_TYPE);
+    assertThat(auditAgent.getAgentUsername()).isEqualTo(AGENT_USERNAME);
+  }
+
+  @Test
+  public void testLogEntryToEventBundleHeader() {
+    final AuditLogEntry logEntry =
+        new AuditLogEntry()
+            .agentId(AGENT_ID)
+            .agentUsername(AGENT_USERNAME)
+            .agentType(USER_AGENT_TYPE)
+            .actionType(ACTION_TYPE_DELETE)
```

Comment:
can also assert for this

Reply:
I couldn't find which one is missing. (Not all fields are set, since some are irrelevant).

Reply:
specifically I mean the actionType - the only one that appears to be missing here

Reply:
It's below. 
 `assertThat(header.getActionType()).isEqualTo(ACTION_TYPE_DELETE);`
It doesn't line up because it's at the top level.

Reply:
doh, thanks